PULL_REQUEST_WORKFLOW.rst

```diff
@@ -0,0 +1,260 @@
+
+2) When new PR is created, only a "default set" of matrix test are running. Only default
+   values for each of the parameters are used effectively limiting it to running matrix builds for only
+   one python version and one version of each of the backends. In this case only one CI and one PROD
+   image is built, saving precious job slots. This build takes around 50% less time than the "full matrix"
+   build.
+
+3) When such PR gets approved, the system further analyses the files changed in this PR and further
+   decision is made what should be communicated to both Committer and Reviewer.
+
+3a) In case of "doc-only" builds, a message is communicated that the PR is ready to be merged and
+    no tests are needed.
+
+.. image:: images/pr/pr-no-tests-needed-comment.png
+    :align: center
+    :alt: No tests needed for "doc-only" builds
+
+3b) In case of "non-core" builds a message is communicated that such PR is likely OK to be merged as is with
+    limited set of tests, but that the committer might decide to re-run the PR after applying
+    "full tests needed" label, which will trigger full matrix build for tests for this PR. The committer
+    might make further decision on what to do with this PR.
+
+.. image:: images/pr/pr-likely-ok-to-merge.png
+    :align: center
+    :alt: Likely ok to merge the PR with only small set of tests
+
+3c) In case of "core" builds (i. e. when the PR touches some "core" part of Airflow) a message is
+    communicated that this PR needs "full test matrix", the "full tests needed" label is applied
+    automatically and either the contributor might rebase the request to trigger full test build or the
+    committer might re-run the build manually to trigger such full test rebuild. Also a check "in-progress"
+    is added, so that the committer realises that the PR is not yet "green to merge". Pull requests with
+    "full tests needed" label always trigger the full matrix build when rebased or re-run so if the
+    PR gets rebased, it will continue triggering full matrix build.
+
+.. image:: images/pr/pr-full-tests-needed.png
+    :align: center
+    :alt: Full tests are needed for the PR
+
+4) If this or another committer disapproves previously approved PR with "full tests needed" label, the
+   bot automatically removes the label, moving it back to "run only default set of parameters" mode.
+   For PRs touching core of airflow once the PR gets approved back, the label will be restored. If it was
+   manually set by the committer, it has to be restored manually.
+
+Note that setting the labels and adding comments might be delayed, due to limitation of Github Actions,
+in case of queues, processing of Pull Request reviews might take some time, so it is advised not to merge
+PR immediately after approval. Luckily, the comments describing the status of the PR trigger notifications
+for the PRs and they provide good "notification" for the committer to act on a PR that was recently
+approved.
```

Comment:
I am happy with this is okay workflow wise, but probably needs to be highlighted in some way. perhaps a `.. note::` block?